nexus/go-handler.md

```diff
@@ -0,0 +1,307 @@
+	Operation   string
+	RequestID   string
+	CallbackURL string
+
+	URL    *url.URL
+	Header http.Header
+	Body   []byte
+}
+
+// Maps almost 1:1 to the Nexus Go SDK CancelOperationRequest, exposing HTTP header and body directly instead of the
+// HTTP Request object.
+type CancelOperationRequest struct {
+	Operation   string
+	OperationID string
+
+	URL    *url.URL
+	Header http.Header
+}
+
+type OperationResponse interface {
+	// private methods
+}
+
+// Indicates that an operation completed successfully. Implements OperationResponse.
+type OperationResponseSync struct {
+	// Header to deliver in the HTTP response.
+	Header http.Header
+	// Body conveying the operation result.
+	// If it is an io.Closer, it will be automatically closed by the framework.
+	// Note that we accept a reader instead of bytes here to keep the future possiblity of supporting streaming of large
+	// payloads.
+	Body io.Reader
+}
+
+// Indicates that an operation has been accepted and will complete asynchronously. Implements OperationResponse.
+type OperationResponseAsync struct {
+	OperationID string
+}
+```
+
+To minimize the amount of boilerplate required to map operations to common Temporal constructs, the SDK exposes default
+handler implementations.
+
+### Workflow Run
+
+```go
+package operation // go.temporal.io/sdk/operation
+
+type WorkflowStartOptions struct {
+	// Options to start the workflow with. TaskQueue defaults to the current worker's task queue.
+	// The workflow ID used here will be returned as the operation ID by default.
+	client.WorkflowStartOptions
+	// Type of workflow to start.
+	WorkflowType string
+	// Input to provide to the workflow.
+	Input        []any
+	// TODO: Should Temporal Header be settable here?
+}
+
+// Returns a Handler that can maps and operation to a workflow that can be started and canceled.
+// Handles HTTP request to payload deserialization for input of type `T`.
+func NewWorkflowRunHandler[T any](func(context.Context, T, *StartOperationRequest) (*WorkflowStartOptions, error)) Handler
+```
+
+**Basic usage:**
+
+```go
+w.RegisterOperation("provision-cell", NewWorkflowRunHandler(
+	func(ctx context.Context, input Input, request *StartOperationRequest) (*WorkflowStartOptions, error) {
+		return &WorkflowStartOptions{
+			Input:        []any{input},
+			WorkflowType: "provision-cell",
+			StartWorkflowOptions: StartWorkflowOptions{
+				ID: fmt.Sprintf("%s-%s", request.Operation, input.CellID),
+			},
+		}, nil
+	}))
+```
+
+### Workflow Query
+
+```go
+package operation // go.temporal.io/sdk/operation
+
+type WorkflowQueryOptions struct {
+	WorkflowID           string
+	// Optionally expose RunID too
+	QueryType            string
+	QueryRejectCondition enums.QueryRejectCondition
+	// Input to provide to the query.
+	Input                []any
+	// TODO: Should Temporal Header be settable here?
+}
+
+func NewWorkflowQueryHandler[T any](func(context.Context, T, *StartOperationRequest) (*WorkflowQueryOptions, error)) Handler
```

Comment:
Mentioned above, I don't think it's too much to ask users to make the client calls themselves